test/test_rhat.py

```diff
@@ -33,11 +41,117 @@ def test_rhat_ragged():
     np.testing.assert_allclose(rhat_expected(chains), rhat(chains), atol=0.1, rtol=0.2)
 
 
-def test_rhat_exceptions():
-    with pt.raises(ValueError):
-        chains = []
-        rhat(chains)
-    with pt.raises(ValueError):
-        chain1 = [1.01, 1.2, 1.3, 1.4]
-        chains = [chain1]
-        rhat(chains)
+def test_rhat_size_exceptions():
+    bad1 = []
+    bad2 = ([[1.01, 1.2, 1.3, 1.4]],)
+    bad3 = [[1, 2, 3], [4], [5, 6, 7, 8, 9]]
+    for chains in [bad1, bad2, bad3]:
+        with pt.raises(ValueError):
+            rhat(chains)
```

Comment:
This style of test should be written to use [`pytest.mark.parameterize`](https://docs.pytest.org/en/7.3.x/how-to/parametrize.html)

```suggestion
@pt.mark.parametrize("chains", [[],  ([[1.01, 1.2, 1.3, 1.4]],),  [[1, 2, 3], [4], [5, 6, 7, 8, 9]]])
def test_rhat_size_exceptions(chains):
    with pt.raises(ValueError):
        rhat(chains)
```

Reply:
Syntax permitting, I'd further suggest that this is easier to read if you give each example on its own line & add a comment about what it's testing ("at least two chains required", "at least two chains required", "Every chain must have at least two elements")

Reply:
I find the `parametrize` [sic] syntax really clunky but made the changes as suggested.   I find the misspelling particularly annoying.

Reply:
Apparently it is an acceptable 'alternative' spelling in American english, though I also wish there was another e.

The primary alternative to `parametrize` is just splitting the tests into different functions, which has the nice benefit that the function names can serve as those comments @jsoules recommended.

In case it wasn't clear, the benefit of doing this versus the for loop approach is specificity. When you use `parametrize`, pytest reports each different parameter value as a unique "test" in the output, so if one of them fails you can immediately see which one. In the for-loop approach, if any of them fail you just get notified that the entire test failed.